amplify/backend/function/fetchMotionState/src/index.js

```diff
@@ -1,10 +1,42 @@
-const { getLatestMotionState } = require('./utils');
+const { MotionState } = require('@colony/colony-js');
+const {
+  getLatestMotionState,
+  updateStakerRewardsInDB,
+  getMotionData,
+} = require('./utils');
+
 /**
  * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
  */
 exports.handler = async (event) => {
-  const { motionId, colonyAddress } = event.arguments?.input || {};
+  const { motionId, colonyAddress, transactionHash } =
+    event.arguments?.input || {};
   /* Get latest motion state from chain */
   const motionState = await getLatestMotionState(colonyAddress, motionId);
+  /*
+   * Check if we need to update staker rewards
+   * This ensures the rewards are present in the event a motion fails before going to a vote,
+   * so the side that has objected can reclaim their stake.
+   */
+  if (motionState === MotionState.Failed) {
```

Comment:
This is clearly a side effect and makes it less obvious what the `fetchMotionState` lamda is doing...

However, the side effect is idempotent (it will only populate staker rewards once, for those that have not been populated previously), which mitigates the risk of hard-to-debug behaviour.

It's also simpler in the sense that it saves me calling a second lamda to perform the check every time we fetch the motion state.

So, I would argue this is ok to live here, despite the fact the lamda name now isn't perfect. I could, of course, change it, to something like "fetchMotionStateAndCheckStakerRewards"... 

But only if we think that's absolutely necessary as it's a mouthful.